cmd/controller-manager/app/controllermanager.go

```diff
@@ -583,6 +583,8 @@ func setupControllers(mgr controllerruntime.Manager, opts *options.Options, stop
 			ResourceInterpreter: resourceInterpreter,
 			RESTMapper:          mgr.GetRESTMapper(),
 			EventRecorder:       mgr.GetEventRecorderFor("dependencies-distributor"),
+			ConcurrentDependenciesDistributorBindingSyncs:  opts.ConcurrentDependenciesDistributorBindingSyncs,
+			ConcurrentDependenciesDistributorResourceSyncs: opts.ConcurrentDependenciesDistributorResourceSyncs,
```

Comment:
We can reuse the `--concurrent-resource-template-syncs` and `--concurrent-resourcebinding-syncs` because the required concurrency depends on the number of `RB` and `resource templates`.

```suggestion
			ConcurrentDependenciesDistributorBindingSyncs:  opts.ConcurrentResourceBindingSyncs,
			ConcurrentDependenciesDistributorResourceSyncs: opts.ConcurrentResourceTemplateSyncs,
```
I'm concerned that introducing too many flags would complicate the configuration. For example, if the system holds a huge amount of resource templates, they probably need to set the `--concurrent-dependencies-distributor-resource-syncs`, but they should set the `--concurrent-resource-template-syncs` as well.

Reply:
What do you think? @snowplayfire